src/main/java/EventsAndCommands/ChatEventDistributor.java

```diff
@@ -0,0 +1,74 @@
+package EventsAndCommands;
+
+import java.io.DataInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import EventsAndCommands.FunEvents.GoodBotEvent;
+import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.core.hooks.ListenerAdapter;
+
+import org.reflections.*;
+
+/**
+ * CommandHandler
+ */
+public class ChatEventDistributor extends ListenerAdapter{
+
+    private static ChatEventDistributor instance = null;
+
+    private Set<ChatEventHandler> eventHandlers;
+
+    public static ChatEventDistributor getInstance(){
```

Comment:
This is not thread safe, but JDA is quite multi threaded. Not sure where it is used, but please add *at least* a doc comment.